ui/CMakeLists.txt

```diff
@@ -122,7 +123,7 @@ endif(HAVE_PRIVATE_QT_HEADERS)
 
 target_link_libraries(gammaray_ui
   LINK_PUBLIC ${QT_QTCORE_LIBRARIES} ${QT_QTGUI_LIBRARIES} gammaray_common
-  LINK_PRIVATE gammaray_common_internal gammaray_kitemmodels
+  LINK_PRIVATE gammaray_common_internal gammaray_kitemmodels gammaray_core
```

Comment:
objecttypefilterproxymodel.h relies on object pointers, so that has to be in core (use in ui would not work), ObjectId based code can work on both sides, so common is better. 

The separation prevents us from accidentally using unsafe (ie. dereferencing pointers to objects in the target process) probe code in the client.